devito/operator.py

```diff
@@ -404,7 +434,7 @@ def apply(self, **kwargs):
         self.cfunction(*arg_values)
 
         # Post-process runtime arguments
-        args = self._postprocess_arguments(args)
+        self._postprocess_arguments(args)
```

Comment:
note for reviewers: This essentially makes sure that SparseFunctions do a gather upon returning from C-land. It only has an effect when running with MPI

Reply:
From my test, this fails with user provided arguments as it only gatheres in `self.output` not in `apply(rec=myrec)` for example

Reply:
discussed offline with @mloubout -- this is gonna be addressed in `mpi-sparse-duplication` (branch of the next imminent PR)